### PR TITLE
Added IOP (Register|Release)IntrHandler test

### PR DESCRIPTION
**Pull request by jpd002**
I wanted to verify that it's only possible to register one handler for an interrupt line. I've also tried to cover all the error codes that those functions can return.

### CODE REVIEW

File: tests/kernel/iop/interrupt/register.expected [hunk 1]
```diff
@@ -0,0 +1,10 @@
+-- TEST BEGIN
+multiple register:
+  register when no handler is present: 0
+  register when handler is already present: -104
+  handler result: 43690
```
Comment (unknownbrackets) [inline tests/kernel/iop/interrupt/register.expected:5]: Is this number a vcount or something?  Is it consistent between runs?

-[Unknown]

File: tests/kernel/iop/interrupt/register.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	printf("invalid params:\n");
+
+	CpuSuspendIntr(&oldStat);
+	{
+		int invalidIntrLineResult = RegisterIntrHandler(~0, 0, (void *)&intrHandler, NULL);
+		printf("  invalid interrupt line: %d\n", invalidIntrLineResult);
+		
+		int invalidModeResult = RegisterIntrHandler(IOP_IRQ_VBLANK, ~0, (void *)&intrHandler, NULL);
+		ReleaseIntrHandler(IOP_IRQ_VBLANK);
+		printf("  invalid mode line: %d\n", invalidModeResult);
```
Comment (unknownbrackets) [inline tests/kernel/iop/interrupt/register.c]: What's the maximum valid mode?  What if it's not negative but just very big?

-[Unknown]

File: tests/kernel/iop/interrupt/register.c [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+		int nullHandlerResult = RegisterIntrHandler(IOP_IRQ_VBLANK, 0, NULL, NULL);
+		ReleaseIntrHandler(IOP_IRQ_VBLANK);
```
Comment (unknownbrackets) [inline tests/kernel/iop/interrupt/register.c:107]: Does the release following a null fail maybe?  Maybe registering null is the same as releasing?

Does registering a null after there's a valid one fail?

-[Unknown]

File: tests/kernel/iop/interrupt/register.expected (hunk 1 above)
Comment (jpd002) [inline tests/kernel/iop/interrupt/register.expected:5]: It's the handler's parameter that comes from here: https://github.com/unknownbrackets/ps2autotests/pull/22/files#diff-43f07e7b7c8b707fd61bdeeb5c40f43bR34. It's consistent between runs.

File: tests/kernel/iop/interrupt/register.c (hunk 2 above)
Comment (jpd002) [inline tests/kernel/iop/interrupt/register.c]: Yes, good point. I can test with a large non-negative value just to see if it returns an error. But, I think the mode parameter is ignored by that function. Every time RegisterIntrHandler is called inside the PS2SDK, the mode is set to 1. Setting it to 0 like, I've done in the tests, doesn't seem to do any difference.

File: tests/kernel/iop/interrupt/register.c (hunk 3 above)
Comment (jpd002) [inline tests/kernel/iop/interrupt/register.c:107]: I'll update the tests to see what the PS2 has to say about these.

**Comment by jpd002** [comment]
Thanks for the comments. The tests should be a bit more complete now.

File: tests/kernel/iop/interrupt/register.expected (hunk 1 above)
Comment (unknownbrackets) [inline tests/kernel/iop/interrupt/register.expected:5]: Ah right, sorry.

-[Unknown]

**Comment by unknownbrackets** [comment]
Cool.  The null handling turned out to be a bit interesting.  Thanks.

-[Unknown]

**Pull request merged**